File: bindings/kotlin/examples/GrpcGetObject.kt
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GrpcClient
import iota_sdk.ObjectId
import iota_sdk.ObjectRequest
import iota_sdk.hexEncode
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GrpcClient.newTestnet()

        val objectId =
            ObjectId.fromHex("0x541b117cac18fb1c07a293db300acd12b05c01fa81232b37151b005ca7d4f755")

        // `objects` is batched: it takes a list of requests and returns the objects
        // in the same order.
        val obj = client.objects(listOf(ObjectRequest(objectId)))[0]

        println("Object ID: ${obj.id().toHex()}")
        println("Version: ${obj.version()}")
        println("Previous transaction: ${obj.previousTransaction().toBase58()}")
        println("Owner: ${obj.owner()}")
        println("Storage rebate: ${obj.storageRebate()}")
        println("Type: ${obj.objectType()}")
        println("BCS bytes: ${hexEncode(obj.asStruct().contents)}")
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}

File: bindings/python/examples/grpc_get_object.py
# Copyright (c) 2026 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk import *

import asyncio


async def main():
    client = GrpcClient.new_testnet()

    object_id = ObjectId.from_hex(
        "0x541b117cac18fb1c07a293db300acd12b05c01fa81232b37151b005ca7d4f755")

    # `objects` is batched: it takes a list of requests and returns the objects
    # in the same order.
    obj = (await client.objects([ObjectRequest(object_id=object_id)]))[0]

    print("Object ID:", obj.id().to_hex())
    print("Version:", obj.version())
    print("Previous transaction:", obj.previous_transaction().to_base58())
    print("Owner:", obj.owner())
    print("Storage rebate:", obj.storage_rebate())
    print("Type:", obj.object_type())
    print("BCS bytes:", hex_encode(obj.as_struct().contents))


if __name__ == "__main__":
    asyncio.run(main())
